lib/curly/scanner.rb

```diff
@@ -16,9 +16,11 @@ class Scanner
 
     COMMENT_MARKER = /!/
     CONTEXT_BLOCK_MARKER = /@/
-    CONDITIONAL_BLOCK_MARKER = /#/
+    CONDITIONAL_BLOCK_MARKER = /(#if |#)/
+    ELSE_BLOCK_MARKER = /else}}/
```

Comment:
Unfortunately, yes. This is because if we parse it as any other token, the scanner has problem telling the difference between `{{else}}` and a normal method with the else substring in it, like `{{elsemethod}}`. Actually one of the consequences of this PR would be that `else` will be a special word: any presenter with an `else` method would break. But it's unavoidable if we want to have an `{{else}}` token.